modules/rgbd/src/hash_tsdf.hpp

```diff
@@ -115,19 +115,76 @@ class HashTSDFVolumeCPU : public HashTSDFVolume
        VolumeIndex lastVolIndex;
 };
 
-template<typename T>
-Ptr<HashTSDFVolume> makeHashTSDFVolume(const VolumeParams& _volumeParams)
+#ifdef HAVE_OPENCL
+
+typedef cv::Mat _VolumeUnitIndexSet;
+typedef cv::Mat _VolumeUnitIndexes;
+
+class HashTSDFVolumeGPU : public HashTSDFVolume
 {
-    return makePtr<T>(_volumeParams);
-}
+public:
```

Comment:
`zFirstMemOrder` should be set to false here.
Please check that this works faster

Reply:
zFirstMemOrder = true

[ RUN      ] Perf_HashTSDF.integrate
[ PERFSTAT ]    (samples=72   mean=256.35   median=262.84   min=214.20   stddev=28.48 (11.1%))
[ RUN      ] Perf_HashTSDF.raycast
[ PERFSTAT ]    (samples=72   mean=191.22   median=189.84   min=181.73   stddev=5.48 (2.9%))


zFirstMemOrder = false

[ RUN      ] Perf_HashTSDF.integrate
[ PERFSTAT ]    (samples=72   mean=239.37   median=226.67   min=213.40   stddev=25.51 (10.7%))
[ RUN      ] Perf_HashTSDF.raycast
[ PERFSTAT ]    (samples=72   mean=191.31   median=190.30   min=182.12   stddev=5.27 (2.8%))

Reply:
false zFirstMemOrder works faster